hivemind/optim/collaborative.py

```diff
@@ -31,8 +31,8 @@ class CollaborationState:
     def ready_for_step(self):
         return self.samples_accumulated >= self.target_batch_size or get_dht_time() >= self.eta_next_step
 
-    def register_step(self):
-        self.optimizer_step += 1
+    def register_step(self, local_step: int):
+        self.optimizer_step = max(local_step, self.optimizer_step)
```

Comment:
Should we actually do anything below if local_step<self.optimizer_step?

Reply:
yep, even if the local step was out of sync, we must reset current samples_accumulated as they "no longer count"

Reply:
Understood!